tracks/r/exercises/anagram/mentoring.md

```diff
@@ -0,0 +1,64 @@
+### Reasonable solution
+
+This solution is clear and straight-forward.
+
+```r
+anagram <- function(subject, candidates) {
+  
+  subject_letters <- tolower(strsplit(subject, "")[[1]])
+  target <- sort(subject_letters)
+
+  is_anagram <- function(word) {
+    word_letters <- tolower(strsplit(word, "")[[1]])
+    identical(sort(word_letters), target) && !identical(word_letters, subject_letters)
+  }
+  
+  matches <- sapply(candidates, is_anagram)
+
+  if(!any(matches)) {
+    c()
+  } else {
+    candidates[matches]
+  }
+}
+```
+
+The learning goal here is common string operations and checking conditions on whole objects vs. on a per item level.
+
+
+### Common suggestions
+
+- Depending on what the student chooses to do, explain how `identical()` differs from `==`; or how `all()` and `any()` can be used in combination with comparison operators to arrive at the same solution. This can help if students are stuck on why they're not getting a single Boolean back from comparisons on vectors.
+- If the solution looks like the one above, consider discussing the pros and cons of having a nested function (e.g. `is_anagram`) vs. one that is independent from `anagram()`. Remember that the `apply()` function-family can take in additional parameters like `target` and `subject_letters` if the applied function needs to receive more than one parameter.
+
+
+### Talking points
+
+- A different flavor of solution:
+
+```r
+split_sort <- function(x) {
+  paste(sort(unlist(strsplit(tolower(x), ""))), collapse="")
+}
+
+anagram <- function(subject, candidates) {
+  not_equiv <- !grepl(subject, candidates, ignore.case=TRUE)
+  is_ana <- sapply(candidates, split_sort) == split_sort(subject)
+
+  result <- candidates[not_equiv & is_ana]
+  
+  if (identical(result, character(0))) {
+    c()
+  } else {
+    result
+  }
+}
+```
+
+... The use of `grepl()` is encouraged for its performance benefits. Otherwise, most of the same logic is applied in this solution as above.
+
+
+### Concepts
```

Comment:
This info is duplicated. Since it also appears on https://exercism.io/my/tracks/r please consider removing each "concepts" section.

Reply:
I know it's duplicated, but I think it's still important as a reference. I'd rather have it accessible while viewing the solution than having a second tab open just to see the concepts on the R track view.

Reply:
That's understandable, but as a team of maintainers we should keep the maintenance load low by avoiding duplication. Saving a second tab is IMHO not worth risking to forget to update these sections if [`config.json`](https://github.com/exercism/r/blob/master/config.json) is updated.

I'm not against _showing_ the concepts in the mentor notes, but it should then be done via the templating, so that the up-to-date concept list if automatically pulled from the config file. That could probably be implemented somewhere in https://github.com/exercism/website/search?q=mentor-exercise-notes+h1&unscoped_q=mentor-exercise-notes+h1

Reply:
Ah! Now that'd be perfect.